Source/WebKit/UIProcess/WebFrameProxy.cpp

```diff
@@ -461,11 +461,11 @@ void WebFrameProxy::getFrameInfo(CompletionHandler<void(FrameTreeNodeData&&)>&&
     }
 }
 
-FrameTreeCreationParameters WebFrameProxy::frameTreeCreationParameters() const
+FrameTreeCreationParameters WebFrameProxy::frameTreeCreationParameters(std::optional<WebCore::FrameIdentifier> remoteMainFrameIdentifier, std::optional<WebCore::ProcessIdentifier> remoteProcessIdentifier) const
 {
     return {
-        m_frameID,
-        m_process->coreProcessIdentifier(),
+        remoteMainFrameIdentifier ? *remoteMainFrameIdentifier : m_frameID,
```

Comment:
This doesn't look right.  Why are we using a different frame identifier here?
Maybe we should have some remote frame information adjacent to the FrameTreeCreationParameters, but not thrown into it like this.